app/src/main/kotlin/at/bitfire/davdroid/ui/AppTheme.kt

```diff
@@ -30,9 +31,9 @@ fun AppTheme(
     val view = LocalView.current
     if (!view.isInEditMode) {
         SideEffect {
-            val window = (view.context as Activity).window
-            window.statusBarColor = colorScheme.primary.toArgb()
-            WindowCompat.getInsetsController(window, view).isAppearanceLightStatusBars = darkTheme
+            val activity = view.context as? AppCompatActivity
+            val systemBarStyle = SystemBarStyle.auto(Color.TRANSPARENT, Color.TRANSPARENT) { darkTheme }
+            activity?.enableEdgeToEdge(systemBarStyle, systemBarStyle)
         }
```

Comment:
Is this some boilterplate code? Can I find it somewhere?

Or do I have to think about what it really does ("edit mode" etc)?

Reply:
This seems to be the recommended way: https://developer.android.com/develop/ui/compose/layouts/insets#compose-apis But there is nothing on `isInEditMode`.

Reply:
Edit mode is just because when using preview, there isn't an activity attached, so context is not an activity, it's some virtual thing Android Studio creates for the preview. We are conditionally casting it, so there should be no issues. However, there isn't any reason to use edge-to-edge on preview, so I think disabling it should be the way to go.

Reply:
That makes sense! Maybe there should be a comment on that?

Reply:
> And I guess we do the enableEdgeToEdge() here in the AppTheme, so we have it one place only and don't need to do it in every activity?

Yup, that was also an option, but since we already have this wrapper, I used that one.

Also, I did it here in order to have dark mode feedback.